### PR TITLE
Basic Reagent Allergen System

### PR DESCRIPTION
**Pull request by Geevies**
* Added a minor and major blueberry allergy to the disability tab. This will add Anaphylactic Cocktail to your bloodstream, and is counteracted with adrenalin.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Comment by Acetrea** [comment]
I really like this and I'd love to see more things added than just blueberries!

File: code/__HELPERS/global_lists.dm [hunk 1]
```diff
@@ -196,6 +196,8 @@ GLOBAL_LIST_EMPTY(contained_clothing_species_adaption_cache)
 	paths = subtypesof(/datum/character_disabilities)
 	for(var/path in paths)
 		var/datum/character_disabilities/T = new path()
+		if(T.abstract_type == path)
```
Comment (FluffyGhoster) [inline code/__HELPERS/global_lists.dm:199]: Use the appropriate proc instead

**Review by Arrow768 — CHANGES_REQUESTED** [review]
As discussed on discord this is too simplistic.

* A allergen implementation should implement all (or most) currently known allergens that are relevant and not just blueberries 
* A reagent needs to be able to contain more than one allergen
* Chemical reactions should (by default) pass the allergens present in all their `required_reagents` along to the `result`
* It must be possible to override this behaviour in the /datum/chemical_reaction.
* (opt) Ideally it should be possible to have different allergens with different effects.
* (opt) Allergens should be identified by the chemistry analyzer

**Comment by Geevies** [comment]
don't have the time or energy to work on this one again at the moment, feel free to take over if you're interested, i might some time in the future...............

**Pull request closed without merging**